File: app/src/main/java/com/example/architechturestartercode/datasource/local/MoviesDao.java
package com.example.architechturestartercode.datasource.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.architechturestartercode.model.Movie;

import java.util.List;

@Dao
public interface MoviesDao {
    @Insert(onConflict = OnConflictStrategy.ABORT)
    void insertMovies(Movie movie);

    @Delete
    void deleteMovies(Movie movie);

    @Query("SELECT * FROM movies")
    LiveData<List<Movie>> getAllMovies();
}

File: app/src/main/java/com/example/architechturestartercode/data/movie/datasource/local/MoviesDao.kt
package com.example.architechturestartercode.data.movie.datasource.local

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.architechturestartercode.data.movie.datasource.local.model.LocalMovie
import com.example.architechturestartercode.data.movie.datasource.remote.model.RemoteMovie

@Dao
interface MoviesDao {
    @Insert(onConflict = OnConflictStrategy.ABORT)
    suspend fun insertMovies(localMovie: LocalMovie)

    @Delete
    suspend fun deleteMovies(localMovie: LocalMovie)

    @Query("SELECT * FROM movies")
    fun getAllMovies(): LiveData<List<LocalMovie>>
}

